lib/pluginmanager/util.rb

```diff
@@ -1,12 +1,35 @@
 require "rubygems/package"
 
 module LogStash::PluginManager
+
+  class Settings
```

Comment:
ok. you can bring your proposal in a separate issue/PR. also note that this also probably relates to #2631